### PR TITLE
Update deployment scripts to support AWSI automation tests on Linux

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: Junbo Liang <68558268+junbo75@users.noreply.github.com>

- Verified the scripts and automation tests locally
- Ran on Jenkins (Jobs failed due to the IAM role permission issue Allie is looking into)

### CODE REVIEW

**Review by shirangj — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/AWS/CMakeLists.txt [hunk 1]
```diff
@@ -12,12 +12,6 @@
 ################################################################################
 
 if(PAL_TRAIT_BUILD_TESTS_SUPPORTED AND PAL_TRAIT_BUILD_HOST_TOOLS)
-    # Only enable AWS automated tests on Windows
-    set(SUPPORTED_PLATFORMS "Windows" "Linux")
-    if (NOT "${PAL_PLATFORM_NAME}" IN_LIST SUPPORTED_PLATFORMS)
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/CMakeLists.txt:17]: Is there any danger of the tests running on Mac, Mobile etc with this removed?

File: AutomatedTesting/Gem/PythonTests/AWS/common/aws_utils.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 class AwsUtils:
 
     def __init__(self, arn: str, session_name: str, region_name: str):
-        local_session = boto3.Session(profile_name='default')
+        local_session = boto3.Session()
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/common/aws_utils.py:19]: Surprised there aren't changes to the README.md over how to setup your credentials? Esp if we previously just took the default profile.

File: AutomatedTesting/Gem/PythonTests/AWS/common/constants.py [hunk 3]
```diff
@@ -6,11 +6,12 @@
 """
 
 import os
+import platform
 
 # ARN of the IAM role to assume for retrieving temporary AWS credentials
 ASSUME_ROLE_ARN = os.environ.get('ASSUME_ROLE_ARN', 'arn:aws:iam::645075835648:role/o3de-automation-tests')
 # Name of the AWS project deployed by the CDK applications
-AWS_PROJECT_NAME = os.environ.get('O3DE_AWS_PROJECT_NAME', 'AWSAUTO')
+AWS_PROJECT_NAME = (os.environ.get('BRANCH_NAME', '') + '-' + os.environ.get('PIPELINE_NAME', '') + '-' + platform.system()).upper()
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/common/constants.py:14]: minor: Danger here is that you will blow out resource names, they will either be too long or all have BRANCH-PIPELINE and get truncated at the important part of the resource name.

More a concern to be careful about rather than actual issue.

File: AutomatedTesting/Gem/PythonTests/AWS/CMakeLists.txt (hunk 1 above)
Comment (junbo75) [reply AutomatedTesting/Gem/PythonTests/AWS/CMakeLists.txt:17]: Mobile shouldn't be an concern since we check the PAL_TRAIT_BUILD_HOST_TOOLS flag and mobile platforms will be ignored. These tests will run on Mac with this change but I think we do want to the platform is supported. We don't define the jobs for Mac and it won't run during the nightly build.

File: AutomatedTesting/Gem/PythonTests/AWS/common/aws_utils.py (hunk 2 above)
Comment (junbo75) [reply AutomatedTesting/Gem/PythonTests/AWS/common/aws_utils.py:19]: Sure. I can update README for the AWS credentials.

File: AutomatedTesting/Gem/PythonTests/AWS/common/constants.py (hunk 3 above)
Comment (junbo75) [reply AutomatedTesting/Gem/PythonTests/AWS/common/constants.py:14]: I believe that we addressed some of the problems in our own CDK applications. I've tested with a long name locally and the tests passed. Should be fine for the currently Jenkins pipeline setup.

**Review by lmbr-pip — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/AWS/common/constants.py (hunk 3 above)
Comment (onecent1101) [reply AutomatedTesting/Gem/PythonTests/AWS/common/constants.py:14]: Minor: what is the reason behind changing project name to include branch and pipeline name? as we have removed O3DE_AWS_PROJECT_NAME from the jenkins pipeline parameters, then why not just make project name simple? (considering stack gets delete after test)
if so, then you don't even need to use BRANCH_NAME and PIPELINE_NAME these two parameters as well.

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 4]
```diff
@@ -20,16 +20,18 @@
 3. Open a new CLI window at the engine root and set the following environment variables:
     * Windows
         ```
-            Set O3DE_AWS_PROJECT_NAME=AWSAUTO
             Set O3DE_AWS_DEPLOY_REGION=us-east-1
             Set ASSUME_ROLE_ARN=arn:aws:iam::{your_aws_account_id}:role/o3de-automation-tests
+            Set BRANCH_NAME={your_branch_name}
```
Comment (allisaurus) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:25]: I'm not sure what "branch name" and "pipeline name" mean when I'm running this locally. Branch = git branch, I assume? But what should pipeline be?

**Review by lmbr-pip — APPROVED** [review]
LG

**Review by onecent1101 — APPROVED** [review]

File: scripts/build/Platform/Linux/deploy_cdk_applications.sh [hunk 5]
```diff
@@ -80,11 +80,14 @@ fi
...
-AWS_SESSION_TOKEN=$(echo "$credentials" | cut -d' ' -f2)
-AWS_ACCESS_KEY_ID=$(echo "$credentials" | cut -d' ' -f3)
+export AWS_SECRET_ACCESS_KEY=$(echo $credentials | cut -d' ' -f1)
+export AWS_SESSION_TOKEN=$(echo $credentials | cut -d' ' -f2)
+export AWS_ACCESS_KEY_ID=$(echo $credentials | cut -d' ' -f3)
 
-O3DE_AWS_DEPLOY_ACCOUNT=$(echo "$ASSUME_ROLE_ARN" | cut -d':' -f5)
+export O3DE_AWS_DEPLOY_ACCOUNT=$(echo "$ASSUME_ROLE_ARN" | cut -d':' -f5)
+if [[ -z "$O3DE_AWS_PROJECT_NAME" ]]; then
+   export O3DE_AWS_PROJECT_NAME=$BRANCH_NAME-$PIPELINE_NAME-Linux
```
Comment (onecent1101) [inline scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: Minor: change requires here? if branch name and pipeline name are empty

Comment (junbo75) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: Deployment will fail if branch name and pipeline name are empty and cdk will throw an error about the invalid project. These are Jenkins environment variables shared by all the Jenkins job.

Comment (onecent1101) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: I remember for local testing, you expect to run this script as well?

Comment (junbo75) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: Yeah, for local testing, our instruction is to set the O3DE_AWS_PROJECT_NAME environment variable and doesn't mention about the Jenkins parameters (BRANCH_NAME and PIPELINE_NAME) at all. As Allie commented before, these two environment variables are confusing for local testing. My thought here is that we use O3DE_AWS_PROJECT_NAMEif it exists (We can set it on Jenkins as well if required in the future) and use $BRANCH_NAME-$PIPELINE_NAME-Platform as the backup AWS project name which is only for Jenkins.

Comment (onecent1101) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: sounds good to me, just want to confirm behavior, as previously you don't need to set this env variable for local testing as well, it will use AWSAUTO if variable is empty?

Comment (junbo75) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: Previously user need to set O3DE_AWS_PROJECT_NAME=AWSAUTO for local testing. our python test used AWSAUTO as the  default value if this environment variable is not set.

Comment (onecent1101) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: sorry, it confuses me now.
based on previous code, python test tries to get O3DE_AWS_PROJECT_NAME, if this env variable doesn't exist, it will use AWSAUTO as default value. Then why we have to set O3DE_AWS_PROJECT_NAME for local testing?

Comment (junbo75) [reply scripts/build/Platform/Linux/deploy_cdk_applications.sh:89]: Synced offline: the previous O3DE_AWS_PROJECT_NAME is used to make sure that AWSI automation tests use the stacks deployed by the CDK applications.
